core/devmode/src/main/java/io/quarkus/dev/RuntimeCompilationSetup.java

```diff
@@ -32,6 +32,10 @@ public static RuntimeUpdatesProcessor setup(DevModeContext context) throws Excep
         if (!context.getModules().isEmpty()) {
             ServiceLoader<CompilationProvider> serviceLoader = ServiceLoader.load(CompilationProvider.class);
             List<CompilationProvider> compilationProviders = new ArrayList<>();
+            for (CompilationProvider provider : serviceLoader) {
+                compilationProviders.add(provider);
```

Comment:
This looks like the compilation providers are being added twice, as the line below the loop also adds them

Reply:
You're right, I fixed it by removing that line at the end of loop. Thanks for having found this.